components/director/internal/open_resource_discovery/validation.go

```diff
@@ -685,6 +693,16 @@ func isValidDate(date *string) validation.RuleFunc {
 	}
 }
 
+func isAPIPartOfBundles(partOfConsumptionBundles []*model.ConsumptionBundleReference) validation.RuleFunc {
+	return func(value interface{}) error {
+		var err error
+		if partOfConsumptionBundles != nil && len(partOfConsumptionBundles) > 0 {
+			err = errors.New("api without entry points can not be part of consumption bundle")
```

Comment:
Minor: You can return the error here and nil below. 

It will be more readable from my PoV. 

If you think this is better I am ok with it also.

Reply:
I prefer your approach as well.